src/gobupload/storage/relate.py

```diff
@@ -452,6 +439,24 @@ def check_relations(src_catalog_name, src_collection_name, src_field_name):
 """
     _query_missing(dangling, f"{name} dangling destinations")
 
+    _check_relation_table(src_catalog_name, src_collection_name, src_field_name, f"{name} relations table out of sync")
+
+
+def _check_relation_table(src_catalog_name, src_collection_name, src_field_name, log_name, max_warnings=50):
+    checker = RelationTableChecker()
+    errors = checker.check_relation(src_catalog_name, src_collection_name, src_field_name)
+
+    for error in errors[:50]:
+        logger.warning(log_name, {
+            'id': log_name,
+            'data': {'src_id': error},
+        })
+
+    if errors:
+        reported_only = f"Reported only first {max_warnings} warnings." if len(errors) > max_warnings else ""
```

Comment:
why not just if len(errors) > 50: ?
The id of the logger.warnings get the warnings nicely organized.
Only skipping warnings should result in an extra warning message